### PR TITLE
L1-4970: Updated the sdk from additional Properties to customData

### PR DESCRIPTION
**Pull request by babadee08**
https://app.clubhouse.io/venture-leap/story/4970/l1-update-php-sdk-to-use-custom-data

### CODE REVIEW

File: src/Services/AutoMapper/UserConfig.php [hunk 1]
```diff
@@ -77,62 +77,6 @@ function (UserJsonldUserRead $source) {
                 }
             );
 
-        $config->registerMapping(\VentureLeap\UserService\Model\User::class, User::class)
```
Comment (danwer81) [inline src/Services/AutoMapper/UserConfig.php:80]: Is it save to remove this? :-D

File: src/Services/AutoMapper/UserConfig.php [hunk 2]
```diff
@@ -162,7 +106,7 @@ function (User $source) {
                         'active' => $source->isActive(),
                         'roles' => $source->getRoles(),
                         'user_type' => $source->getUserType(),
-                        'additional_properties' => json_encode($source->getAdditionalProperties()),
+                        'custom_data' => $source->getCustomData(),
```
Comment (danwer81) [inline src/Services/AutoMapper/UserConfig.php:109]: Sweet

File: src/Model/User/User.php [hunk 3]
```diff
@@ -168,14 +168,14 @@ public function setUserType(string $userType): void
         $this->userType = $userType;
     }
 
-    public function getAdditionalProperties(): array
```
Comment (danwer81) [inline src/Model/User/User.php:171]: I would love to have functions like `getProperty(string $propertyName)` etc. for convenience (and stability) here.

**Review by danwer81 — APPROVED** [review]
Minor comment, then let's merge.

File: src/Services/AutoMapper/UserConfig.php (hunk 1 above)
Comment (babadee08) [reply src/Services/AutoMapper/UserConfig.php:80]: Yes, the model doesn't exist anymore.

File: src/Model/User/User.php [hunk 4]
```diff
@@ -168,14 +168,33 @@ public function setUserType(string $userType): void
...
     }
 
-    public function setAdditionalProperties(array $additionalProperties): void
+    public function setCustomData(array $customData): void
     {
-        $this->additionalProperties = $additionalProperties;
+        $this->customData = $customData;
+    }
+
+    public function getCustomProperty(string $propertyName)
```
Comment (danwer81) [inline src/Model/User/User.php:181]: Can we have anything else than `string` without breaking the user service? Yeah, right?

File: src/Model/User/User.php [hunk 5]
```diff
@@ -168,14 +168,33 @@ public function setUserType(string $userType): void
...
+    {
+        $customData = $this->getCustomData();
+
+        if (!array_key_exists($propertyName, $customData)) {
```
Comment (danwer81) [inline src/Model/User/User.php:185]: I think this is not necessary if you are using `??` below.

File: src/Model/User/User.php [hunk 6]
```diff
@@ -168,14 +168,33 @@ public function setUserType(string $userType): void
...
+
+        return $customData[$propertyName] ?? null;
+    }
+
+    public function addCustomProperty(string $property, $value): void
+    {
+        $customData = $this->getCustomData();
+        $customData[$property] = $value;
+
+        $this->setCustomData($customData);
```
Comment (danwer81) [inline src/Model/User/User.php:197]: We don't need this :-) I would just do `$this->customData[$property] = $value`

File: src/Model/User/User.php (hunk 5 above)
Comment (danwer81) [reply src/Model/User/User.php:185]: I would just do ` return $this->customData[$property] ?? null`

File: src/Model/User/User.php (hunk 4 above)
Comment (babadee08) [reply src/Model/User/User.php:181]: Nah, the property has to be a string, the only thing we can't guarantee is the return type

**Pull request merged**